chore(cairo_native): add compiler file path configs

avi-starkware opened the pull request.

avi-starkware commented:
Python side:
https://reviewable.io/reviews/starkware-industries/starkware/36600





<!-- Sent from Reviewable.io -->

noaov1 reviewed (changes requested):
> Reviewed 13 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: 13 of 16 files reviewed, 7 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, @meship-starkware, and @Yoni-Starkware)

___
*[`crates/native_blockifier/Cargo.toml` line 49 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRlmq92freK3QlKi8s:-OFRlmq92freK3QlKi8t:b45h) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/Cargo.toml#L49)):*
> ```TOML
> starknet-types-core.workspace = true
> starknet_api.workspace = true
> starknet_sierra_compile.workspace = true
> ```





_Suggestion:_
```TOML
starknet_sierra_compile= {workspace = true, optional = true}
```
___
*[`crates/blockifier/Cargo.toml` line 56 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRhXwt5SBUTvVxfWFM:-OFRhXwu0uaUd2p2ONOe:bh56bct) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/blockifier/Cargo.toml#L56)):*
> ```TOML
> starknet-types-core.workspace = true
> starknet_api.workspace = true
> starknet_sierra_compile.workspace = true
> ```

Why? Don't we want to depend on this crate only when `cairo-native` feature is turned on?



_Code quote:_
```TOML
starknet_sierra_compile.workspace = true
```
___
*[`crates/starknet_sierra_compile/src/config.rs` line 16 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRuPMNCLj_szX7dlSC:-OFRuPMNCLj_szX7dlSD:b-hty4dd) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/starknet_sierra_compile/src/config.rs#L16)):*
> ```XML
>     pub max_casm_bytecode_size: usize,
>     /// Sierra-to-Casm compiler binary path.
>     pub path_to_sierra_to_casm_compiler: Option<PathBuf>,
> ```

No need for now.



_Code quote:_
```XML
    /// Sierra-to-Casm compiler binary path.
    pub path_to_sierra_to_casm_compiler: Option<PathBuf>,
```
___
*[`crates/native_blockifier/src/py_objects.rs` line 169 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRohxdCHcEEWhtl_We:-OFRohxdCHcEEWhtl_Wf:bkzn5tj) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/src/py_objects.rs#L169)):*
> ```XML
>     pub path_to_sierra_to_casm_compiler: Option<String>,
>     pub path_to_sierra_to_native_compiler: Option<String>,
>     pub path_to_libcairo_native_runtime: Option<String>,
> ```

Why `Option`?



_Suggestion:_
```XML
    pub sierra_to_casm_compiler_path: Option<String>,
    pub sierra_to_native_compiler_path: Option<String>,
    pub libcairo_native_runtime_path: Option<String>,
```
___
*[`crates/native_blockifier/src/py_objects.rs` line 170 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRtuF35SZxxLO1bLQV:-OFRtuF35SZxxLO1bLQW:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/src/py_objects.rs#L170)):*
> ```XML
>     pub path_to_sierra_to_native_compiler: Option<String>,
>     pub path_to_libcairo_native_runtime: Option<String>,
>     pub max_casm_bytecode_size: usize,
> ```





_Suggestion:_
```XML
    pub max_bytecode_size: usize,
```
___
*[`crates/native_blockifier/src/py_objects.rs` line 178 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRv5yiE2Tpdta9boan:-OFRv5yiE2Tpdta9boao:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/src/py_objects.rs#L178)):*
> ```XML
>             path_to_sierra_to_casm_compiler: None,
>             path_to_sierra_to_native_compiler: None,
>             path_to_libcairo_native_runtime: None,
> ```





_Suggestion:_
```XML
            path_to_sierra_to_native_compiler: binary_path(out_dir(), CAIRO_NATIVE_BINARY_NAME),,
            path_to_libcairo_native_runtime: None,
```
___
*[`crates/starknet_sierra_compile/src/command_line_compiler.rs` line 85 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRvio114PxNBCF2jp5:-OFRvio114PxNBCF2jp6:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/starknet_sierra_compile/src/command_line_compiler.rs#L85)):*
> ```XML
>         let mut env_vars = vec![];
>         if let Some(path) = &self.config.path_to_libcairo_native_runtime {
>             env_vars.push(("CAIRO_RUNTIME_LIBRARY", path.to_str().unwrap()));
> ```





_Suggestion:_
```XML
env_vars.push(("CAIRO_NATIVE_RUNTIME_LIBRARY", path.to_str().unwrap()));
```


<!-- Sent from Reviewable.io -->

avi-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: 13 of 16 files reviewed, 7 unresolved discussions (waiting on @avivg-starkware, @meship-starkware, @noaov1, and @Yoni-Starkware)

___
*[`crates/starknet_sierra_compile/src/command_line_compiler.rs` line 85 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRvio114PxNBCF2jp5:-OFRxhT41FsH5tYOzzwU:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/starknet_sierra_compile/src/command_line_compiler.rs#L85)):*
> ```XML
>         let mut env_vars = vec![];
>         if let Some(path) = &self.config.path_to_libcairo_native_runtime {
>             env_vars.push(("CAIRO_RUNTIME_LIBRARY", path.to_str().unwrap()));
> ```

Done.


<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/blockifier/Cargo.toml` line 56 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRhXwt5SBUTvVxfWFM:-OFRxudP9Lr8LFfXu0aJ:b-iuwf95) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/blockifier/Cargo.toml#L56)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

Why? Don't we want to depend on this crate only when `cairo-native` feature is turned on?
</blockquote></details>

We need to import the config struct now


<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/native_blockifier/Cargo.toml` line 49 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRlmq92freK3QlKi8s:-OFRy0QEELEAGewH6CCW:b-ju4yq4) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/Cargo.toml#L49)):*
> ```TOML
> starknet-types-core.workspace = true
> starknet_api.workspace = true
> starknet_sierra_compile.workspace = true
> ```

It can't be optional since we need to import the new config struct.


<!-- Sent from Reviewable.io -->

meship-starkware reviewed (commented):
> Reviewed 9 of 16 files at r1, 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: 15 of 19 files reviewed, 7 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, @noaov1, and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

meship-starkware reviewed (commented):
> Reviewed 1 of 16 files at r1, 5 of 6 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: 18 of 19 files reviewed, 9 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, @noaov1, and @Yoni-Starkware)

___
*[`crates/starknet_sierra_compile/src/command_line_compiler.rs` line 85 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRvio114PxNBCF2jp5:-OFaQY9q1-bMVpTRy_FL:b-6ehkv3) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/starknet_sierra_compile/src/command_line_compiler.rs#L85)):*
<details><summary><i>Previously, avi-starkware wrote…</i></summary><blockquote>

Done.
</blockquote></details>

What happens if we don't have a path? Do we assume that the env var is already defined? 
___
*[`config/sequencer/default_config.json` line 147 at r6](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFaOHoc96v5oGacsHoQ:-OFaOHoc96v5oGacsHoR:bv1332) ([raw file](https://github.com/starkware-libs/sequencer/blob/a9f88ff36af743e1c2ae66e22eadb0e0551f7885/config/sequencer/default_config.json#L147)):*
> ```JSON
>     "value": ""
>   },
>   "batcher_config.contract_class_manager_config.compiler_config.path_to_libcairo_native_runtime.#is_none": {
> ```

What does the # mean?



_Code quote:_
```JSON
#is_none":
```
___
*[`config/sequencer/default_config.json` line 262 at r6](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFaOvKi3k5Z5YuEMrtc:-OFaOvKi3k5Z5YuEMrtd:b-lr9f66) ([raw file](https://github.com/starkware-libs/sequencer/blob/a9f88ff36af743e1c2ae66e22eadb0e0551f7885/config/sequencer/default_config.json#L262)):*
> ```JSON
>     "value": 81920
>   },
>   "compiler_config.path_to_libcairo_native_runtime": {
> ```

Why do we need them both as compiled_config and as batcher config?



_Code quote:_
```JSON
  "compiler_config.path_to_libcairo_native_runtime": {
```


<!-- Sent from Reviewable.io -->

meship-starkware reviewed (approved):
> Reviewed 1 of 6 files at r6.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 9 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, @noaov1, and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

avi-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 6 unresolved discussions (waiting on @avivg-starkware, @meship-starkware, @noaov1, and @Yoni-Starkware)

___
*[`config/sequencer/default_config.json` line 262 at r6](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFaOvKi3k5Z5YuEMrtc:-OFaeLdM9HAb2ws0qtQj:b-liehhc) ([raw file](https://github.com/starkware-libs/sequencer/blob/a9f88ff36af743e1c2ae66e22eadb0e0551f7885/config/sequencer/default_config.json#L262)):*
<details><summary><i>Previously, meship-starkware (Meshi Peled) wrote…</i></summary><blockquote>

Why do we need them both as compiled_config and as batcher config?
</blockquote></details>

The batcher config holds the contract class manager config which holds the compiler config.

I forgot to update the default config file after my latest config changes...
Done.
___
*[`crates/native_blockifier/src/py_objects.rs` line 169 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRohxdCHcEEWhtl_We:-OFaLnrZ7gyGhvP5iIcm:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/src/py_objects.rs#L169)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

Why `Option`?
</blockquote></details>

Done.
___
*[`crates/starknet_sierra_compile/src/command_line_compiler.rs` line 85 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRvio114PxNBCF2jp5:-OFafsU94iO6c7QAMlYL:b-s8jqj4) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/starknet_sierra_compile/src/command_line_compiler.rs#L85)):*
<details><summary><i>Previously, meship-starkware (Meshi Peled) wrote…</i></summary><blockquote>

What happens if we don't have a path? Do we assume that the env var is already defined?
</blockquote></details>

It is defined in the build script and in the blockifier.
When running from cargo, the env var should be defined
___
*[`crates/starknet_sierra_compile/src/config.rs` line 16 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRuPMNCLj_szX7dlSC:-OFaLrZr3TZ0_H_PimM6:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/starknet_sierra_compile/src/config.rs#L16)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

No need for now.
</blockquote></details>

Done.
___
*[`crates/native_blockifier/src/py_objects.rs` line 170 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRtuF35SZxxLO1bLQV:-OFaLm5P42QeP0D_GDhT:b39cfb3) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/src/py_objects.rs#L170)):*
> ```XML
>     pub path_to_sierra_to_native_compiler: Option<String>,
>     pub path_to_libcairo_native_runtime: Option<String>,
>     pub max_casm_bytecode_size: usize,
> ```

Done.

Are we sure we want to couple these two sizes?
The default max bytecode size is 80KB for casm
The size of the compiled native in the `starknet_sierra_compile` tests is 12MB


<!-- Sent from Reviewable.io -->

noaov1 reviewed (changes requested):
> Reviewed 3 of 4 files at r5, 4 of 6 files at r6, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 7 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, @meship-starkware, and @Yoni-Starkware)

___
*[`crates/native_blockifier/src/py_objects.rs` line 170 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRtuF35SZxxLO1bLQV:-OFaoPfl6jWh6AZcsW2y:b-tatoiq) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/src/py_objects.rs#L170)):*
<details><summary><i>Previously, avi-starkware wrote…</i></summary><blockquote>

Done.

Are we sure we want to couple these two sizes?
The default max bytecode size is 80KB for casm
The size of the compiled native in the `starknet_sierra_compile` tests is 12MB
</blockquote></details>

Good point! You can name it `max_native_bytecode_size` (we do not deal with casm for now).
___
*[`crates/native_blockifier/src/py_objects.rs` line 178 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRv5yiE2Tpdta9boan:-OFatiiH61muDAFpdBOf:bflseas) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/src/py_objects.rs#L178)):*
<details><summary><i>Previously, avi-starkware wrote…</i></summary><blockquote>

This will not work... These functions are part of the build utils of the starknet_sierra_compile crate.
I think it makes more sense to leave this field empty when the default path should be used.
Alternatively, we can decide to place the artifacts in an easy-to-parse absolute path such as `/tmp` or `/opt`
</blockquote></details>

I don't like it being optional as we will always want to set its value when working with python.
I see that we use default in tests that don't care about this value. You cam set it to `/tmp`. 
___
*[`crates/starknet_sierra_compile/src/config.rs` line 18 at r7](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFapbZPEtPloRknpZAb:-OFapbZPEtPloRknpZAc:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/1da5edb9f3e35fbedeb95e58af06ddbf14fde5a4/crates/starknet_sierra_compile/src/config.rs#L18)):*
> ```XML
>     pub sierra_to_native_compiler_path: Option<PathBuf>,
>     /// Path to Cairo native runtime library file.
>     pub libcairo_native_runtime_path: Option<PathBuf>,
> ```





_Suggestion:_
```XML
    #[cfg(not(feature = "cairo_native"))]
    /// Sierra-to-Native compiler binary path.
    pub sierra_to_native_compiler_path: Option<PathBuf>,
    /// Path to Cairo native runtime library file.
    pub libcairo_native_runtime_path: Option<PathBuf>,
```
___
*[`crates/native_blockifier/src/py_objects.rs` line 169 at r7](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFauMEgAXoqEEk8GMZj:-OFauMEgAXoqEEk8GMZk:b-7zi8ni) ([raw file](https://github.com/starkware-libs/sequencer/blob/1da5edb9f3e35fbedeb95e58af06ddbf14fde5a4/crates/native_blockifier/src/py_objects.rs#L169)):*
> ```XML
>     pub max_bytecode_size: usize,
>     pub sierra_to_native_compiler_path: Option<String>,
>     pub libcairo_native_runtime_path: Option<String>,
> ```

WDYT?



_Suggestion:_
```XML
    #[cfg(not(feature = "cairo_native"))]
    pub sierra_to_native_compiler_path: Option<String>,
    pub libcairo_native_runtime_path: Option<String>,
```
___
*[`crates/starknet_sierra_compile/src/command_line_compiler.rs` line 81 at r7](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFawG894zGi70ATq7yW:-OFawG894zGi70ATq7yX:b-s100p4) ([raw file](https://github.com/starkware-libs/sequencer/blob/1da5edb9f3e35fbedeb95e58af06ddbf14fde5a4/crates/starknet_sierra_compile/src/command_line_compiler.rs#L81)):*
> ```XML
>         let mut env_vars = vec![];
>         if let Some(path) = &self.config.libcairo_native_runtime_path {
>             env_vars.push(("CAIRO_NATIVE_RUNTIME_LIBRARY", path.to_str().unwrap()));
> ```

It is a bit confusing that we have two ways to definr the env vars: one is here and the other one in the config.toml. We might want to remove it from the config toml. Can you please add a TODO?



_Code quote:_
```XML
        if let Some(path) = &self.config.libcairo_native_runtime_path {
            env_vars.push(("CAIRO_NATIVE_RUNTIME_LIBRARY", path.to_str().unwrap()));
```
___
*[`crates/bin/starknet-native-compile/Cargo.lock` line 3 at r7](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFaqdhO6FrNLlBQfCmt:-OFaqdhO6FrNLlBQfCmu:b-ql3ps5) ([raw file](https://github.com/starkware-libs/sequencer/blob/1da5edb9f3e35fbedeb95e58af06ddbf14fde5a4/crates/bin/starknet-native-compile/Cargo.lock#L3)):*
> ```TOML
> # This file is automatically @generated by Cargo.
> # It is not intended for manual editing.
> version = 4
> ```

What triggered the change in this file?



_Code quote:_
```TOML
version = 4
```


<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/bin/starknet-native-compile/Cargo.lock` line 3 at r7](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFaqdhO6FrNLlBQfCmt:-OFb2ragDxK2RSdphcye:bc83l0l) ([raw file](https://github.com/starkware-libs/sequencer/blob/1da5edb9f3e35fbedeb95e58af06ddbf14fde5a4/crates/bin/starknet-native-compile/Cargo.lock#L3)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

What triggered the change in this file?
</blockquote></details>

I rebased and found hundreds of conflicts in the Cargo.lock file, so I deleted it and ran `cargo update`.


<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/native_blockifier/src/py_objects.rs` line 169 at r7](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFauMEgAXoqEEk8GMZj:-OFb3tXx7CDm8QdkAZfk:b-qm9qwq) ([raw file](https://github.com/starkware-libs/sequencer/blob/1da5edb9f3e35fbedeb95e58af06ddbf14fde5a4/crates/native_blockifier/src/py_objects.rs#L169)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

WDYT?
</blockquote></details>

You mean without the `not` right?

Anyway, I don't think we can do that, because this is a python object. We don't have the `cairo_native` feature flag on the python side.


<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/starknet_sierra_compile/src/command_line_compiler.rs` line 81 at r7](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFawG894zGi70ATq7yW:-OFb4UPx21wodlna9-cs:bxkdy5s) ([raw file](https://github.com/starkware-libs/sequencer/blob/1da5edb9f3e35fbedeb95e58af06ddbf14fde5a4/crates/starknet_sierra_compile/src/command_line_compiler.rs#L81)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

It is a bit confusing that we have two ways to definr the env vars: one is here and the other one in the config.toml. We might want to remove it from the config toml. Can you please add a TODO?
</blockquote></details>

It needs to be in the `config.toml` for tests that compile to native without going through the `starknet-native-compile` binary.


<!-- Sent from Reviewable.io -->

meship-starkware reviewed (commented):
> Reviewed 1 of 2 files at r7.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 6 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, @noaov1, and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

meship-starkware reviewed (approved):
> Reviewed 1 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 6 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, @noaov1, and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

noaov1 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 4 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, @meship-starkware, and @Yoni-Starkware)

___
*[`crates/starknet_sierra_compile/src/command_line_compiler.rs` line 85 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRvio114PxNBCF2jp5:-OFbX-pp5sDkvUkqHugQ:b-hri7jy) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/starknet_sierra_compile/src/command_line_compiler.rs#L85)):*
<details><summary><i>Previously, avi-starkware wrote…</i></summary><blockquote>

It is defined in the build script and in the blockifier.
When running from cargo, the env var should be defined
</blockquote></details>

Maybe it's worth documenting it


<!-- Sent from Reviewable.io -->

noaov1 reviewed (approved):
> Reviewed 1 of 6 files at r6, 2 of 2 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 2 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, @meship-starkware, and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/native_blockifier/src/py_objects.rs` line 170 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRtuF35SZxxLO1bLQV:-OFbl6kP9Qe8yWiDbR48:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/src/py_objects.rs#L170)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

Good point! You can name it `max_native_bytecode_size` (we do not deal with casm for now).
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/native_blockifier/src/py_objects.rs` line 169 at r7](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFauMEgAXoqEEk8GMZj:-OFbl8svA1SQ0ivkiCMH:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/1da5edb9f3e35fbedeb95e58af06ddbf14fde5a4/crates/native_blockifier/src/py_objects.rs#L169)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

Yes I meant without the `not`.
And yes, you are right. It is relevant only for the rust (not pyo3 object).
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/native_blockifier/src/py_objects.rs` line 178 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRv5yiE2Tpdta9boan:-OFblAix-Lk1y8_6y5s4:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/native_blockifier/src/py_objects.rs#L178)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

I don't like it being optional as we will always want to set its value when working with python.
I see that we use default in tests that don't care about this value. You cam set it to `/tmp`.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/starknet_sierra_compile/src/command_line_compiler.rs` line 85 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRvio114PxNBCF2jp5:-OFblJUe69RgpBJMca4w:b-xxcdwj) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/starknet_sierra_compile/src/command_line_compiler.rs#L85)):*
<details><summary><i>Previously, noaov1 (Noa Oved) wrote…</i></summary><blockquote>

Maybe it's worth documenting it
</blockquote></details>

Where?


<!-- Sent from Reviewable.io -->

avi-starkware commented:
*[`crates/starknet_sierra_compile/src/config.rs` line 18 at r7](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFapbZPEtPloRknpZAb:-OFblLdKAPc26AZMe_Jr:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/1da5edb9f3e35fbedeb95e58af06ddbf14fde5a4/crates/starknet_sierra_compile/src/config.rs#L18)):*
> ```XML
>     pub sierra_to_native_compiler_path: Option<PathBuf>,
>     /// Path to Cairo native runtime library file.
>     pub libcairo_native_runtime_path: Option<PathBuf>,
> ```

Done.


<!-- Sent from Reviewable.io -->

noaov1 reviewed (approved):
> Reviewed 8 of 8 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 1 unresolved discussion (waiting on @avivg-starkware, @meship-starkware, and @Yoni-Starkware)

___
*[`crates/starknet_sierra_compile/src/command_line_compiler.rs` line 85 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFRvio114PxNBCF2jp5:-OFbo-xu1AnHso5DTE6j:bmpvaiq) ([raw file](https://github.com/starkware-libs/sequencer/blob/d939588eb51c9823c3a97873cf5dcfa2db18e799/crates/starknet_sierra_compile/src/command_line_compiler.rs#L85)):*
<details><summary><i>Previously, avi-starkware wrote…</i></summary><blockquote>

Where?
</blockquote></details>

Above the `if let Some(path) = &self.config.libcairo_native_runtime_path`.


<!-- Sent from Reviewable.io -->

noaov1 reviewed (approved):
> Reviewed 3 of 3 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 1 unresolved discussion (waiting on @avivg-starkware, @meship-starkware, and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

noaov1 reviewed (approved):
> Dismissed @meship-starkware from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 1 unresolved discussion (waiting on @avivg-starkware and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

noaov1 reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 1 unresolved discussion (waiting on @avivg-starkware and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

noaov1 reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 2 unresolved discussions (waiting on @avi-starkware, @avivg-starkware, and @Yoni-Starkware)

___
*[`crates/starknet_sierra_compile/src/config.rs` line 14 at r13](https://reviewable.io/reviews/starkware-libs/sequencer/3027#-OFlqgKY8A_I4ddZIwc8:-OFlqgKZ8TmzlPHMYmnv:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/8a4af73761f2e97e81d5563e532d146f003c430d/crates/starknet_sierra_compile/src/config.rs#L14)):*
> ```XML
> pub struct SierraCompilationConfig {
>     /// CASM bytecode size limit.
>     pub max_casm_bytecode_size: usize,
> ```





_Suggestion:_
```XML
    /// CASM bytecode size limit (in felts).
    pub max_casm_bytecode_size: usize,
```


<!-- Sent from Reviewable.io -->

meship-starkware reviewed (approved):
> Reviewed 2 of 8 files at r9, 2 of 3 files at r10, 4 of 4 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 1 unresolved discussion (waiting on @avi-starkware, @avivg-starkware, and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

meship-starkware reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: all files reviewed, 1 unresolved discussion (waiting on @avi-starkware, @avivg-starkware, and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->

noaov1 reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/3027)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @avivg-starkware and @Yoni-Starkware)



<!-- Sent from Reviewable.io -->
